src/api/java/com/minecolonies/api/blocks/AbstractBlockHut.java

```diff
@@ -12,13 +12,17 @@
 import com.minecolonies.api.tileentities.AbstractTileEntityColonyBuilding;
 import com.minecolonies.api.tileentities.MinecoloniesTileEntities;
 import com.minecolonies.api.tileentities.TileEntityColonyBuilding;
+import com.minecolonies.coremod.items.ItemBlockHut;
+import com.minecolonies.coremod.research.UnlockBuildingResearchEffect;
```

Comment:
This is introducing a circular dependency. Maybe some of the research stuff can be boosted into the API?

Reply:
The Item is necessary in this class, because this class is the one that needs to register it, also moving the item to the api shouldn't cause any problems.
I am however not sure how to handle `UnlockBuildingResearchEffect`, as it would be possible to not use it in here by moving the `checkResearch(IColony, String)` method to a utility class.
I don't think there is a utility class where it would fit tho.

Reply:
This should be solved now.